File: tests/fixtures/sample-python-project-positive/services/user_service/permissions.py
"""Permission keys and identifier-style constants.

The string values are stable identifiers used in JWT claims and RPC payloads,
not credentials.
"""

from enum import Enum


class Permission(str, Enum):
    """Permission identifiers used in JWT claims."""

    MANAGE_SECRETS = 'manage_secrets'
    MANAGE_API_KEYS = 'manage_api_keys'
    MANAGE_WEBHOOKS = 'manage_webhooks'
    VIEW_BILLING = 'view_billing'


class ErrorCode(str, Enum):
    """Stable error codes returned in API responses."""

    INVALID_TOKEN = 'INVALID_TOKEN'
    MISSING_TOKEN = 'MISSING_TOKEN'
    EXPIRED_TOKEN = 'EXPIRED_TOKEN'
    INVALID_API_KEY = 'INVALID_API_KEY'


REMOTE_API_KEY = 'app_remote_api'
INVITATION_TOKEN_KEY = 'app_invitation_token'
SESSION_API_KEY_VARIABLE = 'APP_SESSION_API_KEYS_0'
MASKED_API_KEY = '**********'

SERVICE_TOKEN_ENV_VARS = {
    'github': 'GITHUB_TOKEN',
    'gitlab': 'GITLAB_TOKEN',
    'bitbucket': 'BITBUCKET_TOKEN',
}


def build_service_payload(provider: str) -> dict:
    """Build a service-integration payload referencing env-var names."""
    return {
        'provider': provider,
        'tokenEnvVar': SERVICE_TOKEN_ENV_VARS[provider],
    }

File: tests/fixtures/sample-js-project-positive/services/user-service/src/constants/permissions.ts
/**
 * Permission keys and error-code identifiers.
 * The string values are stable identifiers used in JWTs / RPC payloads —
 * they are NOT credentials.
 */

export const Permission = {
  ManageSecrets: 'manage_secrets',
  ManageApiKeys: 'manage_api_keys',
  ManageWebhooks: 'manage_webhooks',
  ViewBilling: 'view_billing',
} as const;
export type Permission = (typeof Permission)[keyof typeof Permission];

export const ErrorCode = {
  InvalidToken: 'INVALID_TOKEN',
  MissingToken: 'MISSING_TOKEN',
  ExpiredToken: 'EXPIRED_TOKEN',
  InvalidApiKey: 'INVALID_API_KEY',
} as const;

export type ErrorCodeValue = (typeof ErrorCode)[keyof typeof ErrorCode];

export const USER_SIGNUP_VERIFICATION_TOKEN_IDENTIFIER = 'confirmation-email';
export const ORGANISATION_ACCOUNT_LINK_VERIFICATION_TOKEN_IDENTIFIER = 'organisation-account-link';
export const INVITATION_TOKEN_KEY = 'app_invitation_token';
export const REMOTE_API_KEY = 'app_remote_api';

export const ServiceTokenEnvVar = {
  GitHub: 'GITHUB_TOKEN',
  GitLab: 'GITLAB_TOKEN',
  Bitbucket: 'BITBUCKET_TOKEN',
} as const;

export const SESSION_API_KEY_VARIABLE = 'APP_SESSION_API_KEYS_0';
